### PR TITLE
Enable style cop

### PR DESCRIPTION
**Pull request by ogaca-dd**
This PR enables style cop.

Note:
- As the PR is quite big, the following points will be addressed in separated PRs: 
   - Enable SA0001: XML comment analysis is disabled due to project configuration
   - Fix style for `StatsdClient.Tests`.
- This PR will be merged when `olivierg/AddTelemetry` will be merged

### CODE REVIEW

File: src/StatsdClient/StatsdClient.csproj [hunk 1]
```diff
@@ -22,5 +23,10 @@
       <Version>4.3.0</Version>
     </PackageReference>
   </ItemGroup>
-
+  <ItemGroup>
```
Comment (truthbk) [inline src/StatsdClient/StatsdClient.csproj:26]: Can we get to run this stuff run in the CI?

File: src/StatsdClient/StatsdUDP.cs [hunk 2]
```diff
@@ -78,42 +84,49 @@ internal static IPAddress GetIpv4Address(string name)
...
             return ipAddress;
         }
 
-        public void Send(string command)
+        private static string GetHostNameFromEnvVar()
         {
-            SocketSender.Send(MaxUDPPacketSize, command, 
-                encodedCommand => UDPSocket.SendTo(encodedCommand, encodedCommand.Length, SocketFlags.None, IPEndpoint));
-
+            return Environment.GetEnvironmentVariable(StatsdConfig.DD_AGENT_HOST_ENV_VAR);
```
Comment (truthbk) [inline src/StatsdClient/StatsdUDP.cs:109]: I noticed we renamed this constant: https://github.com/DataDog/dogstatsd-csharp-client/pull/111/files#diff-d0ebbcc771918be301e58f78757158cbL15,  should `DD_AGENT_HOST_ENV_VAR` be renamed as well?

**Comment by truthbk** [comment]
Also, noticed there are a couple of conflicts, they're both trivial.

File: src/StatsdClient/StatsdUDP.cs (hunk 2 above)
Comment (ogaca-dd) [reply src/StatsdClient/StatsdUDP.cs:109]: I renamed `ENTITY_ID_INTERNAL_TAG_KEY` as this field was private.

`DD_AGENT_HOST_ENV_VAR` is a public field and cannot be renamed without breaking client code. Instead, I marked the field as obsolete in this PR:
https://github.com/DataDog/dogstatsd-csharp-client/pull/113/files#diff-d30af27dd1bd06efdef90392c5163883R48-R55

File: src/StatsdClient/StatsdClient.csproj (hunk 1 above)
Comment (ogaca-dd) [reply src/StatsdClient/StatsdClient.csproj:26]: With the changes done in `src/StatsdClient/StatsdClient.csproj` warnings are generated when compiling.

We should catch StyleCop warnings in AppVeyor as I used `/warnaserror` flag at: https://github.com/DataDog/dogstatsd-csharp-client/pull/111/files#diff-180360612c6b8c4ed830919bbb4dd459R10.

I will test it once previous PRs will be merged (AppVeyor tests are failing).

Comment (ogaca-dd) [reply src/StatsdClient/StatsdClient.csproj:26]: Good catch!

There is an existing issue that prevents `/warnaserror` to work as expected: https://github.com/dotnet/installer/issues/1708.

Comment (ogaca-dd) [reply src/StatsdClient/StatsdClient.csproj:26]: Fixed.
This build https://ci.appveyor.com/project/Datadog/dogstatsd-csharp-client/builds/32467628 is an example of Style Cop issues caught by the CI.

**Pull request merged**